presto-spi/src/main/java/io/prestosql/spi/type/TypeSignature.java

```diff
@@ -36,6 +36,8 @@
     private final List<TypeSignatureParameter> parameters;
     private final boolean calculated;
 
+    private int hashCode;
```

Comment:
```
// Cached hash code. The class remains effectively immutable, because inputs for hash calculations are immutable.
```

Reply:
https://github.com/prestosql/presto/pull/5930

Reply:
This is not accurate. The class is very much not "effectively immutable". The value of `hashCode` *does* change after construction and publication (it goes from `0` to some deterministically computed hashcode). It is also theoretically possible that some thread may never observe a non-zero value.

Reply:
@martint i know. I tried to capture the gist of it, or a hint at this, in a brief comment. I don't thing going into tiny details is needed in a code comment here, but i think it's worth having something. Can you suggest a different wording for this?

Reply:
Just added the following comment.
```
// Cached hash code
```

Reply:
Discussed offline with @findepi, merging without a comment for now to avoid confusion.